nominations/tests.py

```diff
@@ -1,3 +1,87 @@
 from django.test import TestCase
+from models import Application
+from local_groups.models import Group
 
-# Create your tests here.
+
+class ApplicationTestCase(TestCase):
+    def setUp(self):
+        # create group
+        group = Group.objects.create(
+            name='Test Group',
+            group_id=0000
+        )
+
+        # create application
+        Application.objects.create(
+            group=group,
+            rep_email='john@example.com',
+            rep_first_name='John',
+            rep_last_name='Doe',
+            rep_phone='6155555555',
+            candidate_first_name='Candidate',
+            candidate_last_name='Candidateson',
+            candidate_office='Governor',
+            candidate_state='TN'
+        )
+
+    def test_application_creation(self):
+        """
+        When an application is created, a Questionnaire and a Nomination
+        should be created as well.
+        """
+
+        app = Application.objects.first()
+
+        # check for nominations and questionnaire
+        self.assertNotEqual(app.nomination, None)
+        self.assertNotEqual(app.questionnaire, None)
+        self.assertEqual(app, app.nomination.application)
+        self.assertEqual(app, app.questionnaire.application_set.first())
+
+    def test_status_generation(self):
```

Comment:
I suggest to make this a unit test, this test should just focus on the `generate_application_status` method and make sure it returns the correct value based on test set up. Then there should be a separate unit test for `save` method that makes sure the `generate_application_status` method is called at the appropriate times. So this way the **unit** tests are very granular.